### PR TITLE
Improve chat initial experience

### PR DESCRIPTION
**Pull request by mdmohsin7**
Related to #889 
- [x] Initial message should include the initial default questions for the user to try, make them exciting to try.
- [x] Uses the name, to make it more user focused.
- [x] Clear chat option (where to show it?)
- [x]  Lets make the progress when loading chat messages for the first time better
- [x] Let's also try loading chat once the app opens, instead of after a user press on chat and has to wait, do some caching too I guess
- [x] Chat memory detail navigation is better, + errors are handled (could not find any errors except for incorrect memory details showing up due to how the new approach has been structured which is fixed as well. Also made improvements, the memory only gets downloaded if it doesn't exist in the memory_provider. So the navigation should be quick in most cases)

https://github.com/user-attachments/assets/5967181a-1220-4bfd-b5b0-d3f8e915993f

https://github.com/user-attachments/assets/c9b8a3d6-8c8e-440c-9018-90f82af0ee30

https://github.com/user-attachments/assets/a1f25006-9a71-41a5-a74f-c53dd9d42020

### Summary by Entelligence.AI

Release Notes:

- New Feature: Added the ability for users to clear their chat history, improving user control over data.
- New Feature: Introduced an animated mini banner widget that enhances the UI experience.
- Refactor: Optimized memory detail navigation and initial chat options for better performance and user interaction.
- Refactor: Improved message loading and chat clearing behavior in the `MessageProvider` class.
- Bug Fix: Adjusted logic in `addMemoryWithDate` method to correctly handle memory insertion based on date comparison.
- Chore: Added a new backend endpoint to clear chat history and return initial message, enhancing server-side functionality.

### CODE REVIEW

File: backend/database/chat.py [hunk 1]
```diff
@@ -82,3 +83,27 @@ def get_messages(uid: str, limit: int = 20, offset: int = 0, include_memories: b
...
+            break
+        batch = db.batch()
+        for doc in docs_list:
+            batch.delete(doc.reference)
+        batch.commit()
+
+
+async def clear_chat( uid: str):
+    try:
+        user_ref = db.collection('users').document(uid)
```
Comment (josancamon19) [inline backend/database/chat.py:109]: don't use db in a router

File: app/lib/pages/chat/widgets/ai_message.dart [hunk 2]
```diff
@@ -127,30 +129,58 @@ class _AIMessageState extends State<AIMessage> {
...
-                                modifiedDetails.createdAt,
-                                MessageMemoryStructured(
-                                  modifiedDetails.structured.title,
-                                  modifiedDetails.structured.emoji,
-                                ));
-                            widget.message.memories.clear();
-                            widget.message.memories.addAll(copy);
-                            SharedPreferencesUtil().modifiedMemoryDetails = null;
-                            setState(() {});
+                          if (idx != -1) {
```
Comment (josancamon19) [inline app/lib/pages/chat/widgets/ai_message.dart:140]: clean chat history should have confirmation button.

File: backend/database/chat.py (hunk 1 above)
Comment (josancamon19) [reply backend/database/chat.py:109]: You don't .delete, check the other delete in the code, is setting a flag `{deleted: True}`

**Review by josancamon19 — CHANGES_REQUESTED** [review]

**Comment by josancamon19** [comment]
The clear chat might appear top right or top left, when I start scrolling, and disappear when I'm at the bottom of the page

File: backend/database/chat.py (hunk 1 above)
Comment (mdmohsin7) [reply backend/database/chat.py:109]: > don't use db in a router

Yes it's in database directory only

Comment (mdmohsin7) [reply backend/database/chat.py:109]: > You don't .delete, check the other delete in the code, is setting a flag `{deleted: True}`

Got it

**Pull request merged**